logstreams/src/test/java/io/zeebe/logstreams/impl/backpressure/IncreasingClockTest.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.zeebe.logstreams.impl.backpressure;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+public class IncreasingClockTest {
+  private static final int TRIES = 100_000;
+
+  @Test
+  public void shouldReturnStrictlyIncreasingValues() {
+    // given
+    final IncreasingClock clock = new IncreasingClock();
```

Comment:
To make this test really use full I think we need dependency injection here. You pass an supplier in which gives twice the  same value or a lower value and then you can verify that your increasing clock still gives an higher value back.